runtime/src/snapshot_package.rs

```diff
@@ -16,14 +13,73 @@ use solana_sdk::hash::Hash;
 use std::{
     fs,
     path::{Path, PathBuf},
-    sync::mpsc::{Receiver, SendError, Sender},
+    sync::{
+        mpsc::{Receiver, SendError, Sender},
+        Arc, Mutex,
+    },
 };
 use tempfile::TempDir;
 
+/// The sender side of the AccountsPackage channel, used by AccountsBackgroundService
 pub type AccountsPackageSender = Sender<AccountsPackagePre>;
+
+/// The receiver side of the AccountsPackage channel, used by AccountsHashVerifier
 pub type AccountsPackageReceiver = Receiver<AccountsPackagePre>;
+
+/// The error type when sending an AccountsPackage over the channel fails
 pub type AccountsPackageSendError = SendError<AccountsPackagePre>;
 
+/// The PendingSnapshotPackage passes a SnapshotPackage from AccountsHashVerifier to SnapshotPackagerService for archiving
+pub type PendingSnapshotPackage = Arc<Mutex<Option<SnapshotPackage>>>;
+
+/// This enum wraps an AccountsPackage to differentiate between a full snapshot package and an
+/// incremental snapshot package
+pub enum SnapshotPackage {
+    FullSnapshotPackage(AccountsPackage),
+    IncrementalSnapshotPackage(AccountsPackage),
+}
+
+/// This trait provides the `archive_snapshot_package()` function that facilitates archiving a
+/// snapshot package
+pub trait SnapshotPackageArchiver {
+    /// Archive a SnapshotPackage
+    fn archive_snapshot_package(&self, maximum_snapshot_archives_to_retain: usize) -> Result<()>;
+}
+
+impl SnapshotPackageArchiver for SnapshotPackage {
+    fn archive_snapshot_package(&self, maximum_snapshot_archives_to_retain: usize) -> Result<()> {
+        match self {
+            SnapshotPackage::FullSnapshotPackage(accounts_package) => {
+                snapshot_utils::archive_snapshot_package(
+                    accounts_package,
+                    maximum_snapshot_archives_to_retain,
+                    snapshot_utils::TMP_FULL_SNAPSHOT_PREFIX,
+                )
+            }
+            SnapshotPackage::IncrementalSnapshotPackage(accounts_package) => {
+                snapshot_utils::archive_snapshot_package(
+                    accounts_package,
+                    maximum_snapshot_archives_to_retain,
+                    snapshot_utils::TMP_INCREMENTAL_SNAPSHOT_PREFIX,
+                )
+            }
+        }
+    }
+}
```

Comment:
Here is the main new bits. It's got the updated `PendingSnapshotPacakge`, the `SnapshotPackage` enum, and the trait + impl for `SnapshotPackageArchiver`.